lib/fumanchu/compiler.ex

```diff
@@ -3,11 +3,49 @@ defmodule FuManchu.Compiler do
   alias FuManchu.Parser
   alias FuManchu.Generator
 
+  @spec compile(String.t) :: (map -> String.t)
```

Comment:
See my updated comment. If it raises an error probably should rename to `compile!`, yeah?